src/main/groovy/com/vherasymenko/avro/decoder/core/binary/AvroBinaryDecoderService.groovy

```diff
@@ -9,14 +10,15 @@ import org.apache.avro.io.DecoderFactory
 /**
  * The avro binary decoder.
  */
+@Slf4j
 class AvroBinaryDecoderService implements AvroBinaryDecoderPort {
 
     @Override
     void decodeEvent( Schema schema, byte[] event ) {
         def decoder = new DecoderFactory().binaryDecoder( event, null )
         DatumReader<GenericRecord> reader = new GenericDatumReader<>( schema )
-        def document = reader.read( null, decoder )
-        println( 'Decoded document: ' + document )
-        document
+        def reuse = null // if reuse is provided, it will be reinitialized to the given input stream
```

Comment:
Maybe a better name is `const foo DO_NOT_REUSE_STREAM = null`?  Without the comment, I couldn't even guess what `reuse` was for.  Then the code reads `def document = reader.read( DO_NOT_REUSE_STREAM, decoder )`